volume_type_validation/volume_type_validation.py

```diff
@@ -0,0 +1,187 @@
+"""AWS Volume Type Validation interface."""
+import functools
+import operator
+
+import pandas as pd
+from pluck import pluck
+
+
+class VolumeTypeValidationResult:
+    """Validation Result Processing."""
+
+    def __init__(self):
+        """Initialize value that holds Result."""
+        self.invalid_items = {}
+
+    def add(self, source_id, message):
+        """Append conclusive Result message to Cluser Id."""
+        if source_id not in self.invalid_items:
```

Comment:
@tumido I think that doing 
```
if not self.invalid_items.get(source_id):
   ...
```

and the benchmark should be almost the same?

Reply:
@Ladas If you speak about performance comparison of the two, the `if source_id not in self.invalid_items` should be better than `if not self.invalid_items.get(source_id)` since `dict.get` is switching context and moves the call inside a `try/catch` block. So the first should be a bit faster than the later.

However, the best option here would be this definition:
```
class VolumeTypeValidationResult:
    """Validation Result Processing."""

    def __init__(self):
        """Initialize value that holds Result."""
        self.invalid_items = defaultdict(list)

    def add(self, source_id, message):
        """Append conclusive Result message to Cluser Id."""
        self.invalid_items[source_id].append(message)

    def to_dict(self):
        """Convert Result instance to dict."""
        return self.invalid_items
```

Reply:
My thinking is that `if source_id not in self.invalid_items:` is O(n), while using get() should be O(1)

But sure, if `defaultdict` uses c implementation, lets use it like this.